File: solutions/javascript/RansomNote/RansomNote.js
/**
 * @param {string} ransomNote
 * @param {string} magazine
 * @return {boolean}
 */
var canConstruct = function(ransomNote, magazine) {
    for (let i = 0; i < ransomNote.length; i++) {
        let j = magazine.indexOf(ransomNote[i]);
        if (j === -1) {
            return false;
        }
        magazine = magazine.substr(0, j) + magazine.substr(j + 1);
    }
    return true;
};

File: solutions/java/RansomNote/RansomNote.java
class Solution {
    public boolean canConstruct(String ransomNote, String magazine) {
        if (ransomNote.length() > magazine.length()) {
            return false;
        }
        int[] log = new int[26];
        for (char c: ransomNote.toCharArray()) {
            int x = magazine.indexOf(c, log[c-'a']);
            if(x == -1) {
                return false;
            } else {
                log[c-'a'] = x+1;
            }
        }
        return true;
    }
}